v3/src/components/graph/adornments/adornments-store.ts

```diff
@@ -0,0 +1,102 @@
+import { Instance, types } from "mobx-state-tree"
+import { IAdornmentContentInfo, getAdornmentContentInfo, getAdornmentTypes } from "./adornment-content-info"
+import { IAdornmentComponentInfo, getAdornmentComponentInfo } from "./adornment-component-info"
+import { AdornmentModelUnion, IMeasure, PlotTypes, measures } from "./adornment-types"
+import { IAdornmentModel, IUpdateCategoriesOptions } from "./adornment-models"
+import t from "../../../utilities/translation/translate"
+
+interface IMeasureMenuItem {
+  checked: boolean
+  componentContentInfo?: IAdornmentContentInfo
+  componentInfo?: IAdornmentComponentInfo
+  title: string
+  type: string
+  clickHandler?: () => void
+}
+
+export const AdornmentsStore = types.model("AdornmentsStore", {
+    type: "Adornments Store",
+    adornments: types.array(AdornmentModelUnion),
+    showMeasureLabels: false
+  })
+  .volatile(self => ({
+    activeUnivariateMeasures: [] as string[]
+  }))
+  .actions(self => ({
+    toggleShowLabels() {
+      self.showMeasureLabels = !self.showMeasureLabels
+    },
+    addActiveUnivariateMeasure(measure: string) {
+      const activeUnivariateMeasures = [...self.activeUnivariateMeasures]
+      activeUnivariateMeasures.push(measure)
+      self.activeUnivariateMeasures = activeUnivariateMeasures
+    },
+    removeActiveUnivariateMeasure(measure: string) {
+      const activeUnivariateMeasures = [...self.activeUnivariateMeasures]
+      const index = activeUnivariateMeasures.indexOf(measure)
+      if (index > -1) {
+        activeUnivariateMeasures.splice(index, 1)
+      }
+      self.activeUnivariateMeasures = activeUnivariateMeasures
+    },
+    showAdornment(adornment: IAdornmentModel, type: string) {
+      const adornmentExists = self.adornments.find(a => a.type === type)
+      if (adornmentExists) {
+        adornmentExists.setVisibility(true)
+      } else {
+        self.adornments.push(adornment)
+      }
+    },
+    hideAdornment(type: string) {
+      const adornment = self.adornments.find(a => a.type === type)
+      adornment?.setVisibility(false)
+    }
+  }))
+  .views(self => ({
+    getAdornmentsMenuItems(plotType: PlotTypes) {
+      const registeredAdornments = getAdornmentTypes()
+      const measureMenuItems: IMeasureMenuItem[] = []
+      let addedShowMeasureLabels = false
+      measures[plotType].map((measure: IMeasure) => {
+        const { title, type } = measure
+        const checked = !!self.adornments.find(a => a.type === type)?.isVisible
+        const registeredAdornment = registeredAdornments.find(a => a.type === title)
```

Comment:
You can't rely on the fact that the translated title of the menu item will match the model type of the adornment we're interested in -- that only works in English (and isn't necessarily guaranteed there). You either need to add the actual adornment type to the `measures` table (why is the existing `type` for all of the entries currently `adornment`?) or perhaps this is the time to get rid of the `measures` table entirely and build it from the registered adornments.

Reply:
Sorry I missed it, but that `type` property in `measures` should've been removed. I added it as part of an abandoned idea for identifying the "Show Measure Labels" item as something other than an adornment when it was still listed in the `measures` table (i.e. its type would've been something other than `adornment`, like `control`). I dropped that idea after removing "Show Measure Labels" from the list, but forgot to remove the new `type` property. 

I can easily enough change that `type` property to be used for the individual adornment types, though. I did that and changed the line to `const registeredAdornment = registeredAdornments.find(a => a.type === type)` which makes more sense. Thanks.

I had the impression we wanted to keep using the `measures` table for now. It'd be nice to drop it, but that'd mean there won't be any placeholder listings in the menu for the adornments that don't exist yet. Should we wait on that? Or do you think at this point that's not a concern?

Reply:
> I had the impression we wanted to keep using the `measures` table for now. It'd be nice to drop it, but that'd mean there won't be any placeholder listings in the menu for the adornments that don't exist yet. Should we wait on that? Or do you think at this point that's not a concern?

At this point I'd do whatever involves less work, which I think is leaving it in for now.